### PR TITLE
rdmsr: Don't panic on user GPF.

### PR DESCRIPTION
**Pull request by dancrossnyc**
I believe this fixes what David observed.

Signed-off-by: Dan Cross <cross@gajendra.net>

### CODE REVIEW

**Review by gmacd — APPROVED** [review]
This looks fine to me, any reason not to approve and merge?

**Comment by dancrossnyc** [comment]
Some testing would be nice.... :-D

File: sys/src/9/pc/trap.c [hunk 1]
```diff
@@ -616,6 +616,13 @@ extern void rdmsrfail(void);
 static void
 faultgpf(Ureg* ureg, void*)
 {
+	int user = (ureg->cs & 0xFFFF) == UESEL;
```
Comment (rminnich) [inline sys/src/9/pc/trap.c:619]: There's a macro in fns.h
fns.h:#define   userureg(ur) (((ur)->cs & 0xFFFF) == UESEL)
might want to use that.

Comment (dancrossnyc) [reply sys/src/9/pc/trap.c]: Sure. Fixed up a few other locations of that sequence while I was at it; I left one (which is in a large expression also looking for data segments).

**Review by rminnich — APPROVED** [review]

**Pull request merged**